tsdb/db.go

```diff
@@ -517,7 +518,54 @@ func (db *DBReadOnly) loadDataAsQueryable(maxt int64) (storage.SampleAndChunkQue
 			}
 		}
 		opts := DefaultHeadOptions()
-		opts.ChunkDirRoot = db.dir
+
+		// Just a PoC!!
+		// The idea here is to prevent the read head from writing into RW's chunks_head/
+		// We hardlink the completed/immutable chunks into a new dir and hand it to the RO head.
+		originalChecksDir := MmappedChunksDir(db.dir)
+		// TODO: delete at the end
+		// independentChunksDirRoot := "/tmp/pom_ppoc"
+		independentChunksDirRoot, err := os.MkdirTemp("", "data")
+		if err != nil {
+			panic(err)
+		}
+		independentChunksDir := MmappedChunksDir(independentChunksDirRoot)
+		if err := os.MkdirAll(independentChunksDir, 0o777); err != nil {
+			return nil, err
+		}
+		files, err := chunks.ListChunkFiles(independentChunksDir)
+		if err != nil {
+			return nil, err
+		}
+
+		// Copied from head_chunks.go
+		// Remove the last file as it could be mutable
```

Comment:
Are you sure about this? The only mutable chunk is the head chunk, once they are mmapped they are not right?

Reply:
I thought the RO head would append to the latest chunk file until it reaches `MaxHeadChunkFileSize`, but apparently it'll create its own new file.